### PR TITLE
Feat: BigInput

### PR DESCRIPTION
**Pull request by Libao1**

### CODE REVIEW

**Comment by leo-3108** [comment]
Wie ist hier der Status @Libao1?

File: app/shared/components/BigButton.svelte [hunk 1]
```diff
@@ -0,0 +1,64 @@
+<script type="ts" context="module">
+
+export function generateIcon(preferredJourneyMode:string){
+  if(preferredJourneyMode === "fastest") return "north"
+  if(preferredJourneyMode === "least-changes") return "switch_access_shortcut_add"
+  if(preferredJourneyMode === "least-walking") return "directions_walk"
+
+  return ""
+}
```
Comment (finnge) [inline app/shared/components/BigButton.svelte:9]: This should not be part of a general BigButton Component. This should be included from outside

File: app/shared/components/BigButton.svelte [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+</script>
+
+<script type="ts">
+
+  import { createEventDispatcher } from "svelte";
+  import { PreferredJourneyMode } from "~/types";
+
+  export let icon = ""
+  $: hasIcon = icon !== ""
```
Comment (finnge) [inline app/shared/components/BigButton.svelte:19]: actually ok. but this var is only used in one place. The bool could be included there

File: app/shared/components/BigButton.svelte [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+  export let label = "Default Label";
+
+</script>
+<gridLayout class="button-style">
+    <stackLayout orientation="horizontal" on:tap>
+      <label class={hasIcon
+        ? "icon button-withIcon"
+        : "button-withoutIcon" } text="{icon}"/>
```
Comment (finnge) [inline app/shared/components/BigButton.svelte:27]: You can actually render code dynamically

```suggestion
      {#if hasIcon}
      <label class="icon" text="{icon}"/>
      {/endif}
```

File: app/shared/components/BigButton.svelte [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+      }
+
+      .button-withIcon {
+          color: var(--color-primary);
+          padding-right: calc(var(--base-s)*8px);
+      }
+
+      .button-withoutIcon {
+          visibility: hidden;
+      }
```
Comment (finnge) [inline app/shared/components/BigButton.svelte:54]: in combi with up top, this is not needed

```suggestion
```

**Review by finnge — CHANGES_REQUESTED** [review]

File: app/shared/components/BigButton.svelte (hunk 3 above)
Comment (leo-3108) [reply app/shared/components/BigButton.svelte:27]: klappt nicht

Comment (finnge) [reply app/shared/components/BigButton.svelte:15]: Oh sorry endif ist falsch nur /if

**Review by finnge — APPROVED** [review]

**Pull request merged**